src/worldfactory.cpp

```diff
@@ -169,6 +169,26 @@ WORLD *worldfactory::make_new_world( bool show_prompt, const std::string &world_
     return add_world( std::move( retworld ) );
 }
 
+static cata::optional<std::string> prompt_world_name( const std::string &title,
+        const std::string &cur_worldname )
+{
+    string_input_popup popup;
+    popup.max_length( max_worldname_len ).title( title ).text( cur_worldname );
+    std::vector<input_event> evs =
+        inp_mngr.get_input_for_action( "PICK_RANDOM_WORLDNAME", "WORLDGEN_CONFIRM_DIALOG" );
+    if( !evs.empty() ) {
+        popup.description( string_format(
+                               _( "Press [<color_c_yellow>%s</color>] to randomize the world name." ),
+                               inp_mngr.get_keyname( evs.front().get_first_input(), evs.front().type ) ) );
+        popup.callbacks.emplace( evs.front().get_first_input(), []( utf8_wrapper & txt ) {
```

Comment:
I suggest making the text being edited a member variable of `string_input_popup` and let the callback modify it via the popup variable instead, possibly setting a flag to let the query method handle the change if needed.